### PR TITLE
solution

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -2,24 +2,67 @@
...
 
 
 class ChildrenSlideLimitationValidator(SlideLimitationValidator):
-    pass
+    age = IntegerRange(min_amount=4, max_amount=14)
+    height = IntegerRange(min_amount=80, max_amount=120)
+    weight = IntegerRange(min_amount=20, max_amount=50)
+
+    def __init__(self, age: int, weight: int, height: int) -> None:
+        super().__init__(age, weight, height)
```
Comment (ArsenPidhoretskyi) [inline app/main.py]: These lines are not needed. Remove them

File: app/main.py [hunk 2]
```diff
@@ -2,24 +2,67 @@
...
 
 
 class AdultSlideLimitationValidator(SlideLimitationValidator):
-    pass
+    age = IntegerRange(min_amount=14, max_amount=60)
+    height = IntegerRange(min_amount=120, max_amount=220)
+    weight = IntegerRange(min_amount=50, max_amount=120)
+
+    def __init__(self, age: int, weight: int, height: int) -> None:
+        super().__init__(age, weight, height)
```
Comment (ArsenPidhoretskyi) [inline app/main.py]: These lines are not needed. Remove them

File: app/main.py [hunk 3]
```diff
@@ -2,24 +2,67 @@
...
 
 
 class Slide:
-    pass
+    def __init__(
+        self,
+        name: str,
+        limitation_class: SlideLimitationValidator
```
Comment (ArsenPidhoretskyi) [inline app/main.py]: It accepts type, not instance

**Review by ArsenPidhoretskyi — CHANGES_REQUESTED** [review]

File: app/main.py [hunk 4]
```diff
@@ -2,24 +2,63 @@
...
 
 
 class Slide:
-    pass
+    def __init__(
+        self,
+        name: str,
+        limitation_class: (
+            ChildrenSlideLimitationValidator | AdultSlideLimitationValidator
+        )
```
Comment (ArsenPidhoretskyi) [inline app/main.py]: It accepts type of SlideLimitationValidator, not instances

**Review by ArsenPidhoretskyi — CHANGES_REQUESTED** [review]

File: app/main.py [hunk 5]
```diff
@@ -2,24 +2,61 @@
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, instance, name: str) -> None: # noqa
```
Comment (vasylhnatiuk) [inline app/main.py]: noqa? why you need it? delete please

**Review by vasylhnatiuk — CHANGES_REQUESTED** [review]

File: app/main.py (hunk 5 above)
Comment (ghost) [reply app/main.py]: Do you can explain me how write correct type hint for descriptor's methods?

Comment (vasylhnatiuk) [reply app/main.py]: `object ` is a good idea for annotation

**Review by vasylhnatiuk — APPROVED** [review]

**Pull request closed without merging**